pkg/utils/discovery.go

```diff
@@ -64,8 +64,6 @@ func (arch *AvailableArchitecture) GetHash() string {
 	if arch.hash == "" {
 		arch.calculateHash()
 	}
-	arch.mx.Lock()
-	defer arch.mx.Unlock()
 	return arch.hash
```

Comment:
Hmm, so we want read protection, ok

Reply:
The issue is that writing on a sting variable is not guaranteed to be atomic (even if it is in many architectures). Since this may be modified in another thread, we must protect the read operation.

Reply:
then fixing so `calculateHash` returns a value.